src/Groups/types.jl

```diff
@@ -0,0 +1,233 @@
+import Hecke:
+    abelian_group,
+    automorphism_group,
+    center,
+    codomain,
+    cokernel,
+    compose,
+    degree,
+    derived_series,
+    direct_product,
+    domain,
+    elem_type,
+    elements,
+    free_abelian_group,
+    gens,
+    haspreimage,
+    hom,
+    id_hom,
+    image,
+    index,
+    inv!,
+    isabelian,
+    isbijective,
+    ischaracteristic,
+    isconjugate,
+    iscyclic,
+    isinjective,
+    isinvertible,
+    isisomorphic,
+    isnormal,
+    issimple,
+    issurjective,
+    kernel,
+    mul,
+    mul!,
+    ngens,
+    one!,
+    order,
+    perm,
+    preimage,
+    quo,
+    representative,
+    small_group,
+    sub,
+    subgroups
+
+import Base: ==, parent, show
+
+import GAP.GapObj
+
+export PermGroup, PermGroupElem
+export MatrixGroup, MatrixGroupElem
+export PcGroup, PcGroupElem
+export FPGroup, FPGroupElem
+export AutomorphismGroup, AutomorphismGroupElem
+export elem_type
+
+"""
+TODO: document this
+"""
+abstract type Group <: AbstractAlgebra.Group end
```

Comment:
@GDeFranceschi we should rename this back to GAPGroup:
```suggestion
abstract type GAPGroup <: AbstractAlgebra.Group end
```
... and then of course then adapt all places which refer to this...